src/main/java/com/microsoft/sqlserver/jdbc/MaxResultBufferParser.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.lang.management.ManagementFactory;
+import java.text.MessageFormat;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Parser created to parse String value from Connection String to equivalent number of bytes for JDBC Driver to work on.
+ */
+public class MaxResultBufferParser {
+
+    private static final Logger logger = Logger.getLogger("com.microsoft.sqlserver.jdbc.MaxResultBufferParser");
+    private static final String[] PERCENT_PHRASES = {"percent", "pct", "p"};
+    private static final String ERROR_MESSAGE = "maxResultBuffer property is badly formatted: {0}";
+
+    private MaxResultBufferParser() {}
+
+    /**
+     *
+     * Returns number of bytes for maxResultBuffer property
+     *
+     * @param input
+     *        String value for maxResultProperty provided in Connection String
+     * @return 'maxResultBuffer' property as number of bytes
+     * @throws SQLServerException
+     *         Is Thrown when maxResultProperty's syntax is wrong
+     */
+    public static long validateMaxResultBuffer(String input) throws SQLServerException {
+        String numberString;
+        long number = -1;
+
+        // check for null values and empty String "", if so return -1
+        if (StringUtils.isEmpty(input)) {
```

Comment:
Temporary solution would be to change this line to

```java
if (StringUtils.isEmpty(input) || "-1".equalsIgnoreCase(input)) {
```

But we should ask whether it makes sense to allow users to even input something like -1 at all.